src/emqx_client.erl

```diff
@@ -106,6 +109,7 @@
                 ack_timer       :: reference(),
                 retry_interval  :: pos_integer(),
                 retry_timer     :: reference(),
+                response_info   :: iodata(),
```

Comment:
According specs for `def_response`, this could be `function()` or `iodata()`.

However, it is perhaps practically useless to set a fixed (`iodata()`) response for all requests.
Better to call this field `request_handler` then make a spec `fun((request_input()) -> response_payload())` for it.
`request_input()` can be for now just a `binary()` --- the request payload, but can be extended to be a tuple or map in the future for richer features.

Reply:
Also, it's better to allow initializing the client process with a pre-defined handler, instead of calling `def_response`.
Then the API is better to be named`set_request_handler` in case there is a need to swap handler at runtime.